NAPS2.Lib/Dependencies/DownloadController.cs

```diff
@@ -97,80 +53,126 @@ public void Stop()
 
     public event EventHandler? DownloadProgress;
 
-    private void StartNextDownload()
+    private async Task<MemoryStream?> TryDownloadQueueItemAsync(string filename, string url)
```

Comment:
Let's call this method something different than the one below (e.g. TryDownloadFromUrlAsync); overloads should only be for alternative parameter sets.